### ISSUE TITLE
python packages: spack create should automatically convert 'pythonhosted.org...' urls to 'pypi.io...'

### ISSUE THREAD

**Issue by Sinan81** [LLNL/spack#18397]
### Rationale
It would mean one less step for users in creating python packages.

### Description
location of link to tar files in pypi.org goes like `pythonhosted.org...`. For example:
```
https://files.pythonhosted.org/packages/f9/bd/10398e2c2d2070cc8a9c7153abfbd4ddb2895a2c52a32722ab8689e0cc7d/pyPEG2-2.15.2.tar.gz
```
However, in spack `pypi.io` type link is preferred. For example, the pypi.io link for the above example would be:
```
https://pypi.io/packages/source/p/pypeg2/pyPEG2-2.15.2.tar.gz
```
The logic of going from a give pythonhosted type link to pypi.io link is simple. This is just a matter of where and when to modify this url change in spack classes/functions involved in `spack create` call

@adamjstewart

**Comment by adamjstewart** [comment]
This would be fairly easy to do automatically. In the long run, I would like to add a `pypi` variable similar to `url` that knows what to do. See #17587 and #2718 for related work.

**Comment by adamjstewart** [comment]
Closed by #17587
